src/Sylius/Bundle/CoreBundle/Installer/Provider/DatabaseSetupCommandsProvider.php

```diff
@@ -54,16 +67,19 @@ public function getCommands(InputInterface $input, OutputInterface $output, Ques
                 sprintf('The database <info>%s</info> already exists and it has no tables.', $this->getDatabaseName()),
             );
 
-            return $this->getCreateSchemaOrRunMigrationsCommand($outputStyle);
+            return ['doctrine:migrations:migrate' => ['--no-interaction' => true]];
         }
 
-        return $this->getCreateDatabaseWithSchemaCommands($outputStyle);
+        return [
+            'doctrine:database:create' => [],
+            'doctrine:migrations:migrate' => ['--no-interaction' => true],
+        ];
     }
 
     private function isEmptyDatabasePresent(): bool
     {
         try {
-            return 0 === count($this->getSchemaManager()->listTableNames());
+            return 0 === count($this->entityManager->getConnection()->createSchemaManager()->listTableNames());
```

Comment:
I think this `try-catch` exists only due previous `private getSchemaManager()` throwing `\RuntimeException`.

I'm not familiar with `SchemaManager`, but I would assume if either `getConnection()`, `createSchemaManager()` or `listTableNames()` will throw a exception, then we should let it fail and not treat it like empty tables list.

Reply:
I've just read `isEmptyDatabasePresent()` method name, sorry, I guess it answers my question. :sweat_smile: